maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/report/TestSetRunListener.java

```diff
@@ -76,7 +76,7 @@ public TestSetRunListener( ConsoleReporter consoleReporter, FileReporter fileRep
                                StatisticsReporter statisticsReporter, boolean trimStackTrace,
                                boolean isPlainFormat, boolean briefOrPlainFormat )
     {
-        this.consoleReporter = consoleReporter;
+        this.consoleReporter = consoleReporter != null ? consoleReporter : new NullConsoleReporter();
```

Comment:
@britter 
>>does this belong into DefaultReporterFactory.createReporter()
If you mean this code:
`ConsoleReporter consoleReporter = shouldReportToConsole() ? new ConsoleReporter( consoleLogger ) : null;`

Then I would keep the responsibility of creating anew reporter yet in the method `createReporter()`. I am going to fix a bug exactly in this part of code after you. We can still improve the code by small incremental changes.